jest.config.js

```diff
@@ -1,3 +1,72 @@
+// jest.config.js
+const {defaults: tsjPreset} = require('ts-jest/presets');
 module.exports = {
-  preset: 'react-native',
+  ...tsjPreset,
+  preset: 'jest-expo',
+  setupFilesAfterEnv: ['@testing-library/jest-native/extend-expect'],
+  testPathIgnorePatterns: ['<rootDir>/node_modules/', '\\.snap$'],
+  cacheDirectory: '.jest/cache',
+  testEnvironment: 'jsdom',
+  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json', 'node'],
+  globals: {
+    __DEV__: true,
+  },
+  transform: {
+    '^.+\\.jsx$': 'babel-jest',
+    '^.+\\.tsx?$': [
+      'ts-jest',
+      {
+        // TODO: Remove this to log the type mismatch errors later.
+        diagnostics: {
+          exclude: ['**'],
+        },
+        tsconfig: '<rootDir>/tsconfig.spec.json',
+      },
+    ],
+  },
+  // This line should be kept even with nothing to be ignored, otherwise the transform will not take place.
+  // Not quite sure about the reason.
+  testRegex: '(/__tests__/.*|\\.(test|spec))\\.(ts|tsx|js)$',
+  transformIgnorePatterns: [
+    'node_modules/(?!((jest-)?react-native|@react-native(-community)?)|expo(nent)?|@expo(nent)?/.*|@expo-google-fonts/.*|react-navigation|@react-navigation/.*|@unimodules/.*|unimodules|sentry-expo|native-base|react-native-svg)',
+    'node_modules/(?!(@react-native|react-native|react-native-argon2|@react-navigation|react-native-elements|react-native-size-matters|react-native-ratings|expo-constants|base58-universal|@react-native-*|react-native-google-signin|react-native-linear-gradient|expo-camera|base58-universal/*|react-native-*)/).*/',
+  ],
+  setupFiles: [
+    '<rootDir>/__mocks__/svg.mock.js',
+    '<rootDir>/__mocks__/jest-init.js',
+    '<rootDir>/__mocks__/mmkv-db-setup.js',
+    '<rootDir>/__mocks__/react-native.mock.js',
+    '<rootDir>/__mocks__/telemetry-sdk.mock.js',
+    '<rootDir>/__mocks__/expo-constants.mock.js',
+    '<rootDir>/node_modules/react-native-mmkv-storage/jest/mmkvJestSetup.js',
+    '<rootDir>/node_modules/@react-native-community/netinfo/jest/netinfo-mock.js',
```

Comment:
here we are referring node_modules in setupFiles then do we need netInfo in mock folder https://github.com/mosip/inji/pull/1340/files#diff-b49705f2e2ef827d02423b5feba730994fa3ba08d7f39ca464b4648c65826d98